src/Tasks/Microsoft.NET.Build.Tasks/ProjectContext.cs

```diff
@@ -56,10 +64,24 @@ public IEnumerable<LockFileTargetLibrary> GetRuntimeLibraries(IEnumerable<string
                 allExclusionList.UnionWith(privateAssetsExclusionList);
             }
 
-            if(_filterlockFileTarget != null)
+            if(_packagesToBeFiltered != null)
             {
-                IEnumerable<LockFileTargetLibrary> filterLibraries = _filterlockFileTarget.Libraries;
-                Dictionary<string, LockFileTargetLibrary> filterLookup = filterLibraries.ToDictionary(e => e.Name, StringComparer.OrdinalIgnoreCase);
+                var filterLookup = new Dictionary<string, HashSet<PackageIdentity>>();
```

Comment:
Should use StringComparer.OrdinalIgnoreCase for the Dictionary comparer.  Package Ids are case-insensitive.